firewood/src/proof.rs

```diff
@@ -173,34 +179,24 @@ impl<N: AsRef<[u8]> + Send> Proof<N> {
                 if does_not_match {
                     return Ok((None, Nibbles::<0>::new(&[]).into_iter()));
                 }
+                let data = n.chd_encoded().ok_or(ProofError::InvalidData)?;
 
-                let sub_proof = if term {
-                    SubProof {
-                        encoded: data,
-                        hash: None,
-                    }
-                } else {
-                    self.generate_subproof(data)?
-                };
+                let sub_proof = self.generate_subproof(data.to_vec())?;
```

Comment:
Nit: left the same comment below. You might as well call all the methods you need in the same place
```rust
let data = n.chd_encoded().ok_or(ProofError::InvalidData)?.to_vec();
let sub_proof = self.generate_subproof(data)?;
```